src/webgpu/shader/validation/expression/call/builtin/length.spec.ts

```diff
@@ -0,0 +1,91 @@
+const builtin = 'length';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import {
+  TypeF16,
+  TypeF32,
+  VectorType,
+  elementType,
+  kAllFloatScalarsAndVectors,
+  kAllIntegerScalarsAndVectors,
+} from '../../../../../util/conversion.js';
+import { fpTraitsFor, isRepresentable } from '../../../../../util/floating_point.js';
+import { linearRange } from '../../../../../util/math.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import {
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('splat')
```

Comment:
Are non-splat cases coming? Like using `-val, val, -val` where it can't splat?

Reply:
Yeah, will follow up with non-splat cases. Requires more harness work.
I'll also add cases that are guaranteed to fail, so we have some negative tests.

Reply:
Done